src/main/scala/is/hail/annotations/aggregators/RegionValueVirtualDSAggregator.scala

```diff
@@ -0,0 +1,52 @@
+package is.hail.annotations.aggregators
+import is.hail.annotations.RegionValueBuilder
+import is.hail.annotations._
+import is.hail.expr.types.{TArray, TFloat64, TTuple}
+import org.apache.spark.sql.Row
+
+object RegionValueVirtualDSAggregator {
+  val typ = TArray(TTuple(Array(TFloat64(), TFloat64())))
+}
+
+class RegionValueVirtualDSAggregator(xmin: Double, xmax: Double, ymin: Double, ymax: Double) extends RegionValueAggregator {
+  private var boxes = Set.empty[(Int, Int)]
+
+  def seqOp(r: Region, offset: Long, missing: Boolean) {
+    if (!missing) {
+      val row = UnsafeRow.read(TTuple(Array(TFloat64(), TFloat64())), r, offset).asInstanceOf[UnsafeRow]
+      val x = row.getDouble(0)
+      val y = row.getDouble(1)
+      val i = ((x / ((xmax - xmin) / 10)).toInt, (y / ((ymax - ymin) / 10)).toInt)
+      boxes += i
+    }
+  }
+
+  def combOp(agg2: RegionValueAggregator) {
+    boxes = boxes ++ agg2.asInstanceOf[RegionValueVirtualDSAggregator].boxes
+  }
+
+  def result(rvb: RegionValueBuilder) {
+    rvb.startArray(boxes.size)
+    boxes.foreach(i => {
+      val x = i._1 * ((xmax - xmin) / 10)
+      val y = i._2 * ((ymax - ymin) / 10)
```

Comment:
For numeric precision reasons, it is better to write `(i._1 / (xmax - xmin)) * 10`.

Reply:
@cseed I think doing this will put all of the coordinates on a much smaller plot, because (xmax - xmin) will be a large number, and that number would now be divided instead of multiplied.